kevm-pyk/src/tests/integration/test_conformance.py

```diff
@@ -57,66 +43,145 @@ def _assert_exit_code_zero(pattern: Pattern) -> None:
-)
+ALL_BCHAIN_TESTS_CSV: Final = REPO_ROOT / 'tests/all_bchain_tests.csv'
+BCHAIN_TEST_FILES: Final = tuple(BCHAIN_TEST_DIR.glob('**/*.json'))
+ALL_BCHAIN_TESTS: Final = read_csv_file(ALL_BCHAIN_TESTS_CSV)
+BCHAIN_TESTS: Final = tuple(test for test in ALL_BCHAIN_TESTS if test not in EXCLUDED_TESTS)
+REST_BCHAIN_TESTS: Final = tuple(test for test in ALL_BCHAIN_TESTS if test in EXCLUDED_TESTS)
+assert len(ALL_BCHAIN_TESTS) == len(BCHAIN_TESTS) + len(REST_BCHAIN_TESTS)
+
+
+@pytest.fixture(scope='session')
+def gst_data() -> dict[Path, dict[str, Any]]:
+    return {file: json.loads(file.read_text()) for file in BCHAIN_TEST_FILES + VM_TEST_FILES}
+
+
+@pytest.fixture(scope='session')
+def get_gst_data(gst_data: dict[Path, dict[str, Any]]) -> Callable[[Path, str], dict[str, Any]]:
+    def _get_gst_data(test_file: Path, test_id: str) -> dict[str, Any]:
+        return {test_id: gst_data[test_file][test_id]}
+
+    return _get_gst_data
 
 
 @pytest.mark.parametrize(
-    'test_file',
+    'test_file, test_id',
     BCHAIN_TESTS,
-    ids=[str(test_file.relative_to(TEST_DIR)) for test_file in BCHAIN_TESTS],
+    ids=[f'{test_file}:{test_id}' for test_file, test_id in BCHAIN_TESTS],
 )
-def test_bchain(test_file: Path) -> None:
-    _test(test_file, 'SHANGHAI', 'NORMAL', 1, True)
+def test_bchain(get_gst_data: Callable[[Path, str], dict[str, Any]], test_file: Path, test_id: str) -> None:
+    # Given
+    gst_file = TEST_DIR / test_file
+    gst_input = get_gst_data(gst_file, test_id)
+
+    # When
+    res = interpret(gst_input, 'SHANGHAI', 'NORMAL', 1, True, check=False)
+
+    # Then
+    _assert_exit_code_zero(res)
 
 
 @pytest.mark.skip(reason='failing / slow blockchain tests')
 @pytest.mark.parametrize(
-    'test_file',
+    'test_file, test_id',
     REST_BCHAIN_TESTS,
-    ids=[str(test_file.relative_to(TEST_DIR)) for test_file in REST_BCHAIN_TESTS],
+    ids=[f'{test_file}:{test_id}' for test_file, test_id in REST_BCHAIN_TESTS],
 )
-def test_rest_bchain(test_file: Path) -> None:
-    _test(test_file, 'SHANGHAI', 'NORMAL', 1, True)
+def test_rest_bchain(get_gst_data: Callable[[Path, str], dict[str, Any]], test_file: Path, test_id: str) -> None:
+    # Given
+    gst_file = TEST_DIR / test_file
+    gst_input = get_gst_data(gst_file, test_id)
+
+    # When
+    res = interpret(gst_input, 'SHANGHAI', 'NORMAL', 1, True, check=False)
+
+    # Then
+    _assert_exit_code_zero(res)
+
+
+def test_csv_files(gst_data: dict[Path, dict[str, Any]], update_expected_output: bool) -> None:
+    all_bchain_tests = tuple(
+        sorted(
+            {(file.relative_to(TEST_DIR), test_id) for file in BCHAIN_TEST_FILES for test_id in gst_data[file].keys()}
+        )
+    )
+    assert_or_update_csv_file(all_bchain_tests, ALL_BCHAIN_TESTS_CSV, update_expected_output)
+    all_vm_tests = tuple(
+        sorted({(file.relative_to(TEST_DIR), test_id) for file in VM_TEST_FILES for test_id in gst_data[file].keys()})
+    )
```

Comment:
```suggestion
    all_vm_tests = [file.relative_to(TEST_DIR), test_id) for file in VM_TEST_FILES for test_id in gst_data[file]]
```

Then do the sorting in `assert_or_update_csv_file`.